Dockerfile

```diff
@@ -1,31 +1,45 @@
+# syntax=docker/dockerfile:1.3
+
 ARG GO_VERSION=1.17
+ARG GORELEASER_XX_VERSION=1.1.0
 
-FROM golang:${GO_VERSION}-alpine3.14 AS build
+FROM --platform=$BUILDPLATFORM crazymax/goreleaser-xx:${GORELEASER_XX_VERSION} AS goreleaser-xx
+FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine AS base
+COPY --from=goreleaser-xx / /
+RUN apk add --no-cache file git
+WORKDIR /src
 
+FROM base AS build
 ENV GO111MODULE=auto
-ENV DISTRIBUTION_DIR /go/src/github.com/distribution/distribution
-ENV BUILDTAGS include_oss include_gcs
-
-ARG GOOS=linux
-ARG GOARCH=amd64
-ARG GOARM=6
-ARG VERSION
-ARG REVISION
-
-RUN set -ex \
-    && apk add --no-cache make git file
-
-WORKDIR $DISTRIBUTION_DIR
-COPY . $DISTRIBUTION_DIR
-RUN CGO_ENABLED=0 make PREFIX=/go clean binaries && file ./bin/registry | grep "statically linked"
+ENV CGO_ENABLED=0
+ARG TARGETPLATFORM
+ARG PKG="github.com/distribution/distribution/v3"
+ARG BUILDTAGS="include_oss include_gcs"
+ARG GIT_REF
```

Comment:
Is this `ARG` actually used anywhere? (or is this a special var used by go-releaser?)

Reply:
Yes it's used by GoReleaser to generate the versioning. I will add a comment about that for this ARG.